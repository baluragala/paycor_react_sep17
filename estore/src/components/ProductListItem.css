.item {
  border: 1px solid;
  margin: 20px;
  background-color: lightgray;
  padding: 20px;
}
