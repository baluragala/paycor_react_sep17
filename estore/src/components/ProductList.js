import React, { Component } from "react";
import ProductListItem from "./ProductListItem";

class ProductList extends Component {
  constructor(props) {
    super(props);
    this.state = {
      message: "",
      products: [
        {
          id: 1,
          title: "iphone x",
          price: 2000
        },
        {
          id: 2,
          title: "pixel xl",
          price: 1200
        },
        {
          id: 3,
          title: "note 7",
          price: 1000
        },
        {
          id: 4,
          title: "MI note",
          price: 600
        },
        {
          title: "pixel 3",
          price: 1000,
          id: 5
        }
      ]
    };

    setTimeout(() => {
      this.state.products.push({ id: 6, title: "iphone xs", price: 1999 });
      this.state.products[0] = { ...this.state.products[0], price: 888 };
      this.setState({ products: this.state.products });
    }, 5000);

    this.onProductClicked = this.onProductClicked.bind(this);
  }

  onProductClicked(title) {
    this.setState({ message: `${title} was clicked` });
  }

  loadProducts(e) {
    console.log(e.target);
    alert("Load Products");
  }

  render() {
    return (
      <div>
        <h1 onMouseEnter={() => console.log("enter")}>
          Products ({this.state.message})
        </h1>
        <button onClick={this.loadProducts}>Load Products</button>
        <ul>
          {this.state.products.map(p => (
            <ProductListItem
              key={p.id}
              product={p.price !== 1000 ? p : undefined}
              productClicked={this.onProductClicked}
            />
          ))}
        </ul>
      </div>
    );
  }
}

export default ProductList;
