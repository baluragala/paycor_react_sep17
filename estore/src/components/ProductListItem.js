import React from "react";
import "./ProductListItem.css";

// function ProductListItem(props) {
//   let {
//     product: { title, price }
//   } = props;
//   return (
//     <div className="item">
//       <h2>{title}</h2>
//       <h3>{price}</h3>
//     </div>
//   );
// }

class ProductListItem extends React.Component {
  render() {
    let {
      product: { title, price }
    } = this.props;
    return (
      <div className="item">
        <h2>{title}</h2>
        <h3>{price}</h3>
      </div>
    );
  }
}

export default ProductListItem;
