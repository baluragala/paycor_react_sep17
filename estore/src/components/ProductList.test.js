import Enyzme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { ProductList } from "./ProductList";
import { shallowToJson } from "enzyme-to-json";

import React from "react";
import ProductListItem from "./ProductListItem";
Enyzme.configure({
  adapter: new Adapter()
});

describe("ProductList", () => {
  it("should render correctly", () => {
    const props = {
      prds: [
        {
          id: 1,
          title: "mac book",
          price: 2000
        }
      ],
      msg: "",
      isLoading: false,
      getProducts: jest.fn()
    };
    const wrapper = shallow(<ProductList {...props} />);
    expect(wrapper.find(ProductListItem).length).toEqual(1);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
