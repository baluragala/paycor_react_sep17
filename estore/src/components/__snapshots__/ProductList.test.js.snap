// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductList should render correctly 1`] = `
<div>
  <h1
    onMouseEnter={[Function]}
  >
    Products (
     )
  </h1>
  <button
    onClick={[Function]}
  >
    Load Products
  </button>
  <ul>
    <ProductListItem
      key="1"
      product={
        Object {
          "id": 1,
          "price": 2000,
          "title": "mac book",
        }
      }
      productClicked={[Function]}
    />
  </ul>
</div>
`;
